pkg/fanal/types/artifact.go

```diff
@@ -87,9 +87,37 @@ type Repository struct {
 }
 
 type Layer struct {
-	Digest    string `json:",omitempty"`
-	DiffID    string `json:",omitempty"`
-	CreatedBy string `json:",omitempty"`
+	Size          int64    `json:",omitempty"`
+	Digest        string   `json:",omitempty"`
+	DiffID        string   `json:",omitempty"`
+	CreatedBy     string   `json:",omitempty"`
+	OpaqueDirs    []string `json:",omitempty"`
+	WhiteoutFiles []string `json:",omitempty"`
```

Comment:
Do we need these fields?

Reply:
hmm... Looks like I was wrong and there was no request for ArtifactDetail to contain these fields.
So we don't need these files.

Removed in https://github.com/aquasecurity/trivy/pull/8394/commits/a77f47802e0a5bbc9be8c86b5a52c267f4131239

Reply:
After that change, I think we can remove `Empty()` as [lo.IsEmpty](https://github.com/samber/lo?tab=readme-ov-file#isempty) can be used.

Reply:
nice catch.
removed in https://github.com/aquasecurity/trivy/pull/8394/commits/024fb4a08f9cc6a34a76aa7f99af8e12a1a31561